vault.go

```diff
@@ -8,92 +8,132 @@ import (
 	"strings"
 
 	"github.com/hashicorp/vault/api"
+	"sigs.k8s.io/kustomize/v3/pkg/ifc"
+	"sigs.k8s.io/kustomize/v3/pkg/resmap"
+	"sigs.k8s.io/kustomize/v3/pkg/types"
+	"sigs.k8s.io/yaml"
 )
 
-type plugin struct{}
-
-var KVSource plugin
-
-func getToken() (string, error) {
-	t, exists := os.LookupEnv("VAULT_TOKEN")
-	if !exists {
-		tokenPath, exists := os.LookupEnv("VAULT_TOKEN_PATH")
-		if exists == false {
-			return "", errors.New("No vault token and vault token path")
-		}
+type VaultSecret struct {
+	Path    string
+	Key     string
+	Relabel string
```

Comment:
I think `Relabel` is kind of a confusing name. I think either `name` or `secret_name` might be better?

Reply:
The intent was to indicate that this was an optional field only needed when you need to rename the `Key`. It replaces the `@` syntax in the current API. Do you think `SecretName` is clear enough?

Reply:
oh...  maybe `SecretKey`? `name` is used by the secret, and `label` has its own meaning in k8s.